ivy/functional/frontends/paddle/math.py

```diff
@@ -462,7 +462,9 @@ def reciprocal(x, name=None):
     return ivy.reciprocal(x)
 
 
-@with_unsupported_dtypes({"2.5.1 and below": ("float16", "bfloat16")}, "paddle")
+@with_supported_dtypes(
+    {"2.5.1 and below": ("float32", "float64", "int32", "int64")}, "paddle"
+)
```

Comment:
it does support float16 using a gpu, please update the decorator to be device specifc

Reply:
> it does support float16 using a gpu, please update the decorator to be device specifc

Thanks for highlighting it !